### PR TITLE
[CALCITE-2905] Migrate build system to Gradle

### PR DESCRIPTION
**Pull request by vlsi**
This would be more-or-less similar to https://github.com/apache/calcite-avatica/pull/104

### CODE REVIEW

**Comment by vlsi** [comment]
>I was thinking that the update of the Website instructions

The major part of the current `mvn` instructions are related to "release".
I'm inclined to just add "below instructions are stale", and update the list to the proper ones right during the release.

For instance, we can copy&paste the instructions from Avatica release.

> You send a lot of useful information in the various mails and I think some parts from there definitely deserve to be on the Website.

That's my intention as well.

**Comment by zabetak** [comment]
> I'm inclined to just add "below instructions are stale", and update the list to the proper ones right during the release.
Just curious why during the release and not together with this PR?

**Comment by vlsi** [comment]
> Just curious why during the release and not together with this PR?

TL;DR: to save time.

I'm bad at predicting the future, so I suggest to just try the release, see which commands work, and which configuration options are required.
That is exactly the process we used to release Apache JMeter with pretty much the same Gradle build, and it worked.


Have you seen this: https://github.com/vlsi/vlsi-release-plugins/blob/master/plugins/stage-vote-release-plugin/README.md#making-a-release-candidate ?
In my opinion, it covers the two commands that are needed to publish the artifacts.
What if I just put a cross-reference?

On top of that, there's a section on "use of Docker" which, I guess, @F21 could help to fix (e.g. based on the findings from Avatica)

**Comment by zabetak** [comment]
I understand. I was thinking that if the next RM does not have much experience with the release process or Gradle not having any instructions to begin with may be frustrating. The cross-reference you mentioned may be enough to help people get started.
